R/assert.R

```diff
@@ -84,3 +93,71 @@ vec_is <- function(x, ptype = NULL, size = NULL) {
 
   TRUE
 }
+
+#' Is an object a vector?
+#'
+#' @description
+#'
+#' A vector is a collection of values like a logical vector, a factor,
```

Comment:
What if we called this a "data vector"?

Reply:
I like it. Though what is the opposite of a data vector? An implementation vector?

It seems that in vctrs we'll completely hide the implementation, e.g. there'll be no way to slice a linear model without unclassing it first. From that point of view, it doesn't seem useful to make this distinction.

Another distinction would be data vector versus object vectors like lists and data frames. The latter are recursive because they contain objects rather than data.

Reply:
I like object vectors. A data frame is both a data vector and an object vector, depending which way you slice it.